### PR TITLE
21: Adds data header validation to CheckData

### LINKED ISSUE #21: Validate upload
As I provider, if I try to upload a CSV or Excel file that is not formatted according to the template, or if I try to upload a file that is not a CSV or Excel file, I get an error that I must upload a CSV or Excel file formatted according to the template.

First pass - check ~file type and~ headers

### PR DESCRIPTION
**Pull request by jlhogan**
should close #21 

this is v1; eventually to be replaced with editable data table

### CODE REVIEW

File: client/src/containers/CheckData/CheckData.tsx [hunk 1]
```diff
@@ -1,7 +1,90 @@
...
+		if (loading) return;
+		if (error) return;
+		if (data) return;
+
+		setLoading(true);
+		fetch(`/api/reports/${fileUploadId}`, {
+			headers: {
+				authorization: `Bearer ${accessToken}`
+			}
+		})
```
Comment (cailyncodes) [inline client/src/containers/CheckData/CheckData.tsx:34]: If this lands after #68, let's update it to use the API generated code

File: client/src/containers/CheckData/CheckData.tsx [hunk 2]
```diff
@@ -1,7 +1,90 @@
...
+		.catch((_) => {
+			console.error(_);
+			setError("There was an error");
+		})
+		.finally(() => setLoading(false));
+		
+	}, [accessToken, loading, error, data, fileUploadId]);
+
+	console.log("data", data);
```
Comment (cailyncodes) [inline client/src/containers/CheckData/CheckData.tsx:48]: Nit: cleanup

File: client/src/containers/CheckData/CheckData.tsx (hunk 1 above)
Comment (jlhogan) [reply client/src/containers/CheckData/CheckData.tsx:34]: was planning to wait so i could play w that!

Comment (cailyncodes) [reply client/src/containers/CheckData/CheckData.tsx:34]: Don't worry about this -- I need base of this PR for #73. So I can take care of this.

Comment (jlhogan) [reply client/src/containers/CheckData/CheckData.tsx:34]: not sure i understand ^

**Comment by jlhogan** [comment]
think ii'm going to close this in favor of eventual table PR landing

**Pull request closed without merging**